js/plugins/vertexai/src/index.ts

```diff
@@ -71,9 +92,44 @@ export {
   textEmbeddingGecko003,
   textEmbeddingGeckoMultilingual001,
   textMultilingualEmbedding002,
-  VertexAIEvaluationMetricType as VertexAIEvaluationMetricType,
 };
 
+/**
+ * A document retriever function that takes an array of Neighbors from Vertex AI Vector Search query result, and resolves to a list of documents.
+ * Also takes an options object that can be used to configure the retriever.
+ */
+export type DocumentRetriever = (
+  docIds: Neighbor[],
+  options?: z.infer<typeof VVSRetrieverOptionsSchema>
```

Comment:
the general convention we're trying to stick with is if you're defining a zod schema:

```ts
export const FooSchema = z.object(...)
```

then you also define

```ts
export type Foo = z.infer<typeof FooSchema>
```

note the naming convention: `Foo` + `Schema`